ignite/handlers/ema_handler.py

```diff
@@ -0,0 +1,214 @@
+
+              engine.run(...)
+
+    .. versionadded:: 0.5.0
+
+    """
+
+    def __init__(
+        self,
+        model: nn.Module,
+        momentum: float = 0.0002,
+        momentum_warmup: Optional[float] = None,
+        warmup_iters: Optional[int] = None,
+    ) -> None:
+        if momentum_warmup is not None and not 0 < momentum_warmup < 1:
+            raise ValueError(f"Invalid momentum_warmup: {momentum_warmup}")
+        if not 0 < momentum < 1:
+            raise ValueError(f"Invalid momentum: {momentum}")
+        if momentum_warmup is not None and not momentum_warmup <= momentum:
+            raise ValueError(
+                f"momentum_warmup should be less than or equal to momentum, but got "
+                f"momentum_warmup: {momentum_warmup} and momentum: {momentum}"
+            )
+        if warmup_iters is not None and not (isinstance(warmup_iters, int) and warmup_iters > 0):
+            raise ValueError(f"Invalid warmup_iters: {warmup_iters}")
+        if not isinstance(model, nn.Module):
+            raise ValueError(
+                f"model should be an instance of nn.Module or its subclasses, but got"
+                f"model: {model.__class__.__name__}"
+            )
+        self.momentum_warmup = momentum_warmup
+        self.momentum = momentum
+        self.warmup_iters = warmup_iters
+
+        if isinstance(model, nn.parallel.DistributedDataParallel):
+            model = model.module
+        self.model = model
+
+        self.ema_model = deepcopy(self.model)
+        for param in self.ema_model.parameters():
+            param.detach_()
+        self.ema_model.eval()
+
+    def _get_momentum(self, curr_iter: int) -> float:
+        """Get current momentum, `curr_iter` should be 1-based. When `curr_iter = 1`, `momentum =
+        self.momentum_warmup`; when `curr_iter >= self.warmup_iters`, `momentum = self.momentum`"""
+
+        # TODO: use ignite's parameter scheduling, see also GitHub issue #2090
+        if curr_iter < 1:
+            raise ValueError(f"curr_iter should be at least 1, but got {curr_iter}.")
+
+        # no warmup
+        if self.momentum_warmup is None or self.warmup_iters is None:
+            return self.momentum
+
+        denominator = max(1, self.warmup_iters - 1)
+        momentum = self.momentum_warmup + (self.momentum - self.momentum_warmup) * (curr_iter - 1) / denominator
+        return min(self.momentum, momentum)
+
+    def _update_ema_model(self, engine: Engine, name: str) -> None:
+        """Update weights of ema model"""
+        momentum = getattr(engine.state, name)
+        for ema_p, model_p in zip(self.ema_model.parameters(), self.model.parameters()):
+            ema_p.mul_(1.0 - momentum).add_(model_p.data, alpha=momentum)
+        # directly copy the buffers
+        for ema_b, model_b in zip(self.ema_model.buffers(), self.model.buffers()):
+            ema_b.copy_(model_b.data)
```

Comment:
```suggestion
        # directly copy the buffers
        for ema_b, model_b in zip(self.ema_model.buffers(), self.model.buffers()):
            ema_b.data = model_b.data
```
maybe instead of copying we can create a view of the tensor (less expensive).